src/main/java/javaslang/collection/Iterator.java

```diff
@@ -166,6 +167,16 @@ public T next() {
         };
     }
 
+    @Override
+    default Iterator<T> iterator() {
+        return this;
```

Comment:
Oh - CAUTION - we still need ofAll(*java.util.Iterator*) AND ALSO ofAll(Iterable) because they **lift** java.util.Iterator to javaslang.collection.Iterator. (Wish I had my Notebook here, mobile phone is not good for programming...)